src/pages/task/taskTabs/FileTableTab/GroupedFilesTable/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { useMemo, useRef } from "react";
+import {
+  Table,
+  TableHead,
+  HeaderRow,
+  HeaderCell,
+  TableBody,
+  Row,
+  Cell,
+  LGColumnDef,
+  useLeafyGreenTable,
+  HeaderGroup,
+  LeafyGreenTableRow,
+  flexRender,
+} from "@leafygreen-ui/table/new";
+import { Subtitle } from "@leafygreen-ui/typography";
+import { StyledLink } from "components/styles";
+import { Unpacked } from "types/utils";
+import { GroupedFiles } from "../types";
+
+type GroupedFilesFile = Unpacked<GroupedFiles["files"]>;
+interface GroupedFilesTableProps {
+  files: GroupedFilesFile[];
+  taskName?: string;
+}
+const GroupedFilesTable: React.FC<GroupedFilesTableProps> = ({
+  files,
+  taskName,
+}) => {
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+
+  const columns = useMemo<LGColumnDef<GroupedFilesFile>[]>(
+    () => [
+      {
+        accessorKey: "name",
+        header: "Name",
+        size: 60,
+        enableSorting: true,
+        cell: (value) => (
+          <StyledLink
+            href={value.row.original.link}
+            data-cy="file-link"
+            target="_blank"
+          >
+            {value.getValue()}
+          </StyledLink>
+        ),
+      },
+    ],
+    []
+  );
+  const table = useLeafyGreenTable<GroupedFilesFile>({
+    containerRef: tableContainerRef,
+    data: files,
+    columns,
+  });
+
+  const { rows } = table.getRowModel();
+
+  return (
+    <>
+      {taskName && <Subtitle>{taskName}</Subtitle>}
+      <Table ref={tableContainerRef}>
+        <TableHead>
+          {table
+            .getHeaderGroups()
+            .map((headerGroup: HeaderGroup<GroupedFilesFile>) => (
+              <HeaderRow key={headerGroup.id}>
+                {headerGroup.headers.map((header) => (
+                  <HeaderCell key={header.id} header={header}>
+                    {flexRender(
+                      header.column.columnDef.header,
+                      header.getContext()
+                    )}
+                  </HeaderCell>
+                ))}
+              </HeaderRow>
+            ))}
+        </TableHead>
+        <TableBody>
+          {rows.map((row: LeafyGreenTableRow<GroupedFilesFile>) => (
+            <Row key={row.id} row={row}>
+              {row.getVisibleCells().map((cell) => (
+                <Cell key={cell.id}>
+                  {flexRender(cell.column.columnDef.cell, cell.getContext())}
+                </Cell>
+              ))}
+            </Row>
+          ))}
+        </TableBody>
+      </Table>
```

Comment:
If possible I'd love a bit of extra space between the tables; maybe can add a margin under the table so it won't affect tasks that don't have multiple groups of files?